microsoft-edge: Mirror google-chrome derivation

rhysmdnz opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Changed the edge derivation to be closer to a copy of the google chrome derivation. Should make it easier to follow what is going on in that. Most of the changes that need to be made to edge tend to get done in chrome first.

Sourced from: https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/go/google-chrome/package.nix

Have also removed beta and dev for the same reasons they were dropped in chrome. no one's testing them.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

rhysmdnz committed:
microsoft-edge: Mirror google-chrome derivation

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/4677

rhysmdnz commented:
This fixes a crash in recent versions of edge upon opening the dev tools

gepbird commented on pkgs/by-name/mi/microsoft-edge/package.nix:
```diff
@@ -0,0 +1,289 @@
+
+  passthru.updateScript = ./update.py;
+
+  meta = with lib; {
+    homepage = "https://www.microsoft.com/en-us/edge";
+    description = "Web browser from Microsoft";
+    changelog = "https://learn.microsoft.com/en-us/deployedge/microsoft-edge-relnote-stable-channel";
+    license = licenses.unfree;
+    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
+    platforms = [ "x86_64-linux" ];
+    maintainers = with maintainers; [
+      zanculmarktum
+      kuwii
+      rhysmdnz
+    ];
+  };
```
Please set `meta.mainProgram`, order `meta` attrs alphabetically and remove `with lib;`

gepbird commented on pkgs/by-name/mi/microsoft-edge/package.nix:
```diff
@@ -0,0 +1,289 @@
+in
+
+stdenv.mkDerivation rec {
+  inherit pname;
```
This is the only place `pname` is used
```suggestion
  pname = "microsoft-edge";
```

gepbird commented on pkgs/by-name/mi/microsoft-edge/package.nix:
```diff
@@ -0,0 +1,289 @@
+
+let
+
+  pname = "microsoft-edge";
```
```suggestion
```

gepbird commented on pkgs/by-name/mi/microsoft-edge/package.nix:
```diff
@@ -0,0 +1,289 @@
+    runHook preInstall
+
+    appname=msedge
+    dist=stable
```
Maybe the `dist` should be hard coded?
```suggestion
```

SuperSandro2000 replied:
that would simplify the code

gepbird reviewed (changes requested):
Thanks for your PR! I agree with the removal of the non-stable channels. Many of the refactors look cleaner, but I'm not experienced with this derivation to judge all those dependency and build script changes. Review from another maintainer would be very welcome. I'll only leave some nits.

Next time when you make big refactors, please do it in multiple commits, For example a by-name migration and a many lines of formatting changes should be in separate commits. It's harder to review when the diff only says removed every line of microsoft-edge/browser.nix, added every line of microsoft-edge/package.nix.

gepbird commented:
cc @zanculmarktum @kuwii

Daholli commented:
can we also move up to 130.0.2849.1 since that seems to be the new stable now

SuperSandro2000 commented on pkgs/by-name/mi/microsoft-edge/package.nix:
```diff
@@ -0,0 +1,289 @@
+  libexif,
+  pciutils,
+
+  # Additional dependencies according to other distros.
+  ## Ubuntu
+  curl,
+  liberation_ttf,
+  util-linux,
+  wget,
+  xdg-utils,
+  ## Arch Linux.
+  flac,
+  harfbuzz,
+  icu,
+  libopus,
+  libpng,
+  snappy,
+  speechd-minimal,
+  ## Gentoo
+  bzip2,
+  libcap,
```
We should add more useful comments than just copying other distros

rhysmdnz committed:
microsoft-edge: 129.0.2792.65 -> 130.0.2849.46

rhysmdnz committed:
microsoft-edge: Remove beta and dev versions

rhysmdnz committed:
microsoft-edge: move to by-name

rhysmdnz commented on pkgs/by-name/mi/microsoft-edge/package.nix:
```diff
@@ -0,0 +1,289 @@
+  makeWrapper,
+  patchelf,
+  stdenv,
+  stdenvNoCC,
```
Removed the unused deps from chromes darwin build

rhysmdnz commented:
> Thanks for your PR! I agree with the removal of the non-stable channels. Many of the refactors look cleaner, but I'm not experienced with this derivation to judge all those dependency and build script changes. Review from another maintainer would be very welcome. I'll only leave some nits.
> 
> Next time when you make big refactors, please do it in multiple commits, For example a by-name migration and a many lines of formatting changes should be in separate commits. It's harder to review when the diff only says removed every line of microsoft-edge/browser.nix, added every line of microsoft-edge/package.nix.

Agreed, but there's not really a smaller cleaner diff. This was more a copy over https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/go/google-chrome/package.nix and make the relevant changes to get it back to edge. than it was change the edge derivation.

I've split it up a bit more now.

Daholli commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 347836`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>microsoft-edge</li>
  </ul>
</details>
